Sources/Bot/Services/MergeService.swift

```diff
@@ -106,6 +91,195 @@ public final class MergeService {
     }
 }
 
+// MARK: - System types
+
+extension MergeService {
+
+    public enum FailureReason: String, Equatable, Encodable {
+        case conflicts
+        case mergeFailed
+        case synchronizationFailed
+        case checkingCommitChecksFailed
+        case checksFailing
+        case timedOut
+        case blocked
+        case unknown
+    }
+
+    public struct State: Equatable {
+
+        public enum Status: Equatable, Encodable {
+            case starting
+            case idle
+            case ready
+            case integrating(PullRequestMetadata)
+            case runningStatusChecks(PullRequestMetadata)
+            case integrationFailed(PullRequestMetadata, FailureReason)
+
+            internal var integrationMetadata: PullRequestMetadata? {
+                switch self {
+                case let .integrating(metadata):
+                    return metadata
+                default:
+                    return nil
+                }
+            }
+
+            internal var statusChecksMetadata: PullRequestMetadata? {
+                switch self {
+                case let .runningStatusChecks(metadata):
+                    return metadata
+                default:
+                    return nil
+                }
+            }
+
+            enum CodingKeys: String, CodingKey {
+                case status
+                case metadata
+                case error
+            }
+
+            public func encode(to encoder: Encoder) throws {
+                var values = encoder.container(keyedBy: CodingKeys.self)
+                switch self {
+                case .starting:
+                    try values.encode("starting", forKey: .status)
+                case .idle:
+                    try values.encode("idle", forKey: .status)
+                case .ready:
+                    try values.encode("ready", forKey: .status)
+                case let .integrating(metadata):
+                    try values.encode("integrating", forKey: .status)
+                    try values.encode(metadata, forKey: .metadata)
+                case let .runningStatusChecks(metadata):
+                    try values.encode("runningStatusChecks", forKey: .status)
+                    try values.encode(metadata, forKey: .metadata)
+                case let .integrationFailed(metadata, error):
+                    try values.encode("integrationFailed", forKey: .status)
+                    try values.encode(metadata, forKey: .metadata)
+                    try values.encode(error, forKey: .error)
+                }
+
+            }
+        }
+
+        internal let targetBranch: String
+        internal let integrationLabel: PullRequest.Label
+        internal let topPriorityLabels: [PullRequest.Label]
+        internal let statusChecksTimeout: TimeInterval
+        public let pullRequests: [PullRequest]
+        public let status: Status
```

Comment:
[nit] Can we bring these `State` members to the top of the scope (as in style-guide) ?

Reply:
Fixed in 6345029c29e9bd24a03d4a99c4e9eba56931e4b5